main.go

```diff
@@ -0,0 +1,309 @@
+	flag.StringVar(&outputFile, "output-file", "", "Save output into a file")
+	var prettyPrint bool
+	flag.BoolVar(&prettyPrint, "pretty-print", false, "Pretty-print output")
+
+	flag.Parse()
+
+	if releasesURL == "" {
+		return errReleaseIndexMissing
+	}
+
+	releasePath, err := ReleaseURL(releasesURL)
+	if err != nil {
+		return fmt.Errorf("Error with Release Index: %v", err)
+	}
+
+	parsedURL, err := url.Parse(releasePath)
+	if err != nil {
+		return fmt.Errorf("Error while parsing release path: %v", err)
+	}
+
+	var decoder *json.Decoder
+	if parsedURL.Scheme == "" {
+		// It is most likely a local file.
+		releasesMetadataFile, err := os.Open(releasePath)
+		if err != nil {
+			return fmt.Errorf("Error opening file: %v", err)
+		}
+
+		defer releasesMetadataFile.Close()
+		decoder = json.NewDecoder(releasesMetadataFile)
+	} else {
+		resp, err := http.Get(releasePath)
+		if err != nil {
+			return fmt.Errorf("Error while fetching: %v", err)
+		}
+
+		defer resp.Body.Close()
+		decoder = json.NewDecoder(resp.Body)
+	}
+
+	var release Release
+	if err = decoder.Decode(&release); err != nil {
+		return fmt.Errorf("Error while decoding json: %v", err)
+	}
+
+	streamArch := make(map[string]*StreamArch)
+	for arch, releaseArch := range release.Architectures {
+		archContent := releaseToStream(releaseArch, release)
+		streamArch[arch] = &archContent
+	}
+
+	streamMetadata := StreamMetadata{
+		Stream:        release.Stream,
+		Metadata:      Metadata{LastModified: time.Now().UTC().Format(time.RFC3339)},
+		Architectures: streamArch,
+		Updates:       StreamUpdates{Release: release.Release},
+	}
+
+	if overrideFilename != "" {
+		overrideFile, err := os.Open(overrideFilename)
+		if err != nil {
+			return fmt.Errorf("Can't open file %s: %v", overrideFilename, err)
+		}
+		defer overrideFile.Close()
+
+		streamMetadataJSON, err := json.Marshal(&streamMetadata)
+		if err != nil {
+			return fmt.Errorf("Error during Marshal: %v", err)
+		}
+		streamMetadataMap := make(map[string]interface{})
+		if err = json.Unmarshal(streamMetadataJSON, &streamMetadataMap); err != nil {
+			return fmt.Errorf("Error during Unmarshal: %v", err)
+		}
+
+		overrideMap := make(map[string]interface{})
+		overrideDecoder := json.NewDecoder(overrideFile)
+		if err = overrideDecoder.Decode(&overrideMap); err != nil {
+			return fmt.Errorf("Error while decoding: %v", err)
+		}
+
+		streamMetadataInterface := overrideData(streamMetadataMap, overrideMap)
+		streamMetadataMap = streamMetadataInterface.(map[string]interface{})
+
+		// We are doing Marshal and Unmarshal of streamMetadataMap to keep json in ordered way
+		streamMetadataJSON, err = json.Marshal(streamMetadataMap)
+		if err != nil {
+			return fmt.Errorf("Error during Marshal: %v", err)
+		}
+		if err = json.Unmarshal(streamMetadataJSON, &streamMetadata); err != nil {
+			return fmt.Errorf("Error during Unmarshal: %v", err)
+		}
+	}
+
+	// If outputFile option not specified print on stdout
+	if outputFile != "" {
```

Comment:
Minor: could dedupe this logic by having e.g. an `out` variable that you set to either `os.Stdout` or `streamFile`.

Reply:
Thanks, fixed it